app/grandchallenge/components/models.py

```diff
@@ -2046,3 +2046,39 @@ def create_civ_for_file(self, *, ci, current_civ, new_value):
             # if no new value is provided (user selects '---' in dropdown)
             # delete old CIV
             self.values.remove(current_civ)
+
+
+class ValuesForInterfacesMixin:
+    @property
+    def related_item_model(self):
+        raise NotImplementedError
+
+    @cached_property
+    def interfaces_and_values(self):
+        vals = list(
+            self.related_item_model.select_related(
+                "values", "values__interface", "values__image"
+            )
+            .filter(values__interface__slug__isnull=False)
+            .values(
+                "values__interface__slug",
+                "values__id",
+            )
+            .order_by("values__id")
+            .distinct()
+        )
+        interfaces = [x["values__interface__slug"] for x in vals]
+        return set(interfaces), vals
```

Comment:
Please use a named tuple when returning mutiple values and then access them by attribute name.